components/checkbox/src/auro-checkbox-group.js

```diff
@@ -307,6 +317,7 @@ export class AuroCheckboxGroup extends LitElement {
       checkbox.reset();
     });
 
+    this.touched = false;
```

Comment:
You should be setting `this.touched = false` in all component's `reset()` method